### PR TITLE
feat: export coverage data as JSON

### PR DESCRIPTION
**Pull request by zrock-dev**
#### Resolved task/US: ([CS-54 :Coverage JSON generation](https://app.clickup.com/t/868ayzfqg))
* * * *

## What's in this pull request?

Implementation to export CoverageData as JSON file

Sample of JSON output
```
{
  "name": "org/example",
  "reportPackages": [
    {
      "name": "org/example/domain/",
      "reportClass": [
        {
          "name": "org/example/domain/Car",
          "sourceFileName": "Car.java",
          "methods": [
            {
              "name": "<init>",
              "returnType": "()V",
              "statements": [
                {
                  "statementNumber": 3,
                  "covered": true
                }
              ]
            }
          ]
        }
      ]
    },
    {
      "name": "org/example/",
      "reportClass": [
        {
          "name": "org/example/App",
          "sourceFileName": "App.java",
          "methods": [
            {
              "name": "<init>",
              "returnType": "()V",
              "statements": [
                {
                  "statementNumber": 3,
                  "covered": true
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
```

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Implemented unit tests for new feature. Verified whole project is running with `./gradlew test`

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+
+public class JsonExport implements IExport {
+    public static final String EXPORT_PATH = "coverage.json";
+
+    public void populateStatements(
```
Comment (JosuePrado) [inline src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java]: you can create another class that is “populate coverage data” for this method

File: src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            List<Integer> statements, Map<Integer, ReportLine> reportLineMap) {
+        for (Integer statementNumber : statements) {
+            ReportLine lineReport =
+                    ReportLine.builder().statementNumber(statementNumber).isCovered(false).build();
+            reportLineMap.put(statementNumber, lineReport);
+        }
+    }
+
+    public void populateCoveredStatements(
```
Comment (JosuePrado) [inline src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java]: same case, you can create another class that is “populate coverage data” for this method

File: src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java (hunk 1 above)
Comment (zrock-dev) [reply src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java]: Please clarify why is this change required.

File: src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java (hunk 2 above)
Comment (zrock-dev) [reply src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java]: Please clarify why is this change required

File: src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java (hunk 1 above)
Comment (JosuePrado) [reply src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java]: performs a function or action different from the context of the other methods of the class and the purpose of the class itself.

Comment (zrock-dev) [reply src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java]: Alright, the changes have been implemented

File: src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java (hunk 2 above)
Comment (zrock-dev) [reply src/main/java/edu/usb/argosinstrumentation/export/JsonExport.java]: Alright, the changes have been implemented

**Review by JosuePrado — APPROVED** [review]

**Review by alex-paca-jalafund — APPROVED** [review]
Nice work

**Comment by zrock-dev** [comment]
Following latest message ("Acerca de los PR") about merging PRs, and given that this PR has to approve I'm merging this PR.

**Pull request merged**